### PR TITLE
Wiki feature

### PR DESCRIPTION
**Pull request by makhov**
Issue #3.

### CODE REVIEW

**Comment by nounoursheureux** [comment]
Wow, that's a lot of work. It would be nice to post a screenshot so we can see what it looks like :+1: .

**Comment by makhov** [comment]
Ok, I'll do it soon.

**Comment by tboerger** [comment]
Codewise it already looks pretty good, but i have not tested it.

**Comment by lunny** [comment]
So, let's test it and merge it. @fuxiaohei @tboerger

File: models/repo.go [hunk 1]
```diff
@@ -981,13 +1054,13 @@ func GetRepositories(uid int64, private bool) ([]*Repository, error) {
...
 
-	err := sess.Find(&repos, &Repository{OwnerId: uid})
+	err := sess.Where("is_wiki=?", false).Find(&repos, &Repository{OwnerId: uid})
 	return repos, err
 }
 
 // GetRecentUpdatedRepositories returns the list of repositories that are recently updated.
 func GetRecentUpdatedRepositories(num int) (repos []*Repository, err error) {
-	err = x.Where("is_private=?", false).Limit(num).Desc("updated").Find(&repos)
+	err = x.Where("is_private=?", false).Where("is_wiki=?", false).Limit(num).Desc("updated").Find(&repos)
```
Comment (lunny) [inline models/repo.go]: This line should be
```Go
x.Where("is_private=?", false).And("is_wiki=?", false).Limit(num).Desc("updated").Find(&repos)
```

**Comment by slothbag** [comment]
I get issues trying to compile this merged with develop..

makhov\gitea\gogs.go:39: undefined: cmd.CmdGenerate

**Comment by slothbag** [comment]
"New Page" button doesn't work.

Is it possible to create a link to a page that doesn't exist? And have the wiki engine create a blank page for editing when someone clicks that link?

**Comment by makhov** [comment]
@slothbag, looks like your js is cached.

**Comment by tboerger** [comment]
Showing a new page form should be fine

**Comment by lunny** [comment]
Internal link is broken. @makhov

**Comment by makhov** [comment]
Fix links.

**Comment by lunny** [comment]
Hi, anyone have idea? If not, I will merge it. @fuxiaohei

**Comment by slothbag** [comment]
Why does it need the .wiki folder in the repo directory? I would think the ui can extract from the bare repo when needed like readme.md files.

**Comment by makhov** [comment]
I created **wiki** branch for the feature.

**Comment by makhov** [comment]
@slothbag, because we also need to create and edit files. I'm not sure that it's a good idea, but I guess that cloning repo to temp folder every time is worse.

**Comment by tboerger** [comment]
> @slothbag, because we also need to create and edit files. I'm not sure that it's a good idea, but I guess that cloning repo to temp folder every time is worse.

As long as there are only a few pages cloning is not a bottleneck, but as it grows it gets worse. I think it is ok to keep a cloned copy of the repo.

**Comment by lunny** [comment]
Should this pr need to be merged to develop? @makhov

**Comment by tboerger** [comment]
:+1:

**Comment by makhov** [comment]
I think, we can do it.

**Pull request merged**